### PR TITLE
[FOLLOW UP #2192] Improve hinting UI

### PR DESCRIPTION
**Pull request by aadcg**
Sorry again for making a terrible mistke in #2192.

----

I believe we're very close to merge, so let's go through what might be missing or unclear:

-  I deleted the deprecated ANNOTATE-VISIBLE-ONLY-P since there's a new release coming soon. Is this the correct policy?
-  @aartaka what do you suggest doing about 48317740?

### CODE REVIEW

File: source/mode/element-hint.lisp [hunk 1]
```diff
@@ -153,39 +144,35 @@
...
-                                   (add-element-hints :selector selector)))
-                   :after-destructor
-                   (lambda ()
-                     (with-current-buffer buffer
-                       (remove-element-hints))))))
-      (when result
-        (funcall function result))
-      (with-current-buffer buffer
-        (remove-element-hints)))))
+  (let ((buffer (current-buffer)))
```
Comment (aartaka) [inline source/mode/element-hint.lisp:156]: Maybe make the `buffer` binding part of the `when-let` below too? Will look less nested.

File: source/mode/element-hint.lisp [hunk 2]
```diff
@@ -347,70 +334,60 @@ FUNCTION is the action to perform on the selected elements."
...
     (query-hints "Open element with current modes in new buffer"
                  (lambda (result)
                    (mapcar (alex:rcurry #'%follow-hint-with-current-modes-new-buffer buffer)
-                           result))
-                 :multi-selection-p t)))
+                           result)))))
 
 (define-command copy-hint-url ()
-  "Show a set of element hints, and copy the URL of the user inputted one."
+  "Prompt for element hints and save to clipboard."
```
Comment (aartaka) [inline source/mode/element-hint.lisp:384]: Save what?

File: source/mode/prompt-buffer.lisp [hunk 3]
```diff
@@ -36,6 +36,12 @@ Actions can be listed and run with `return-selection-over-action' (bound to
        "C-down" 'select-last
        "C-pagedown" 'select-next-source
        "C-pageup" 'select-previous-source
+       ;; scroll-page-{down|up}-other-buffer and scroll-other-buffer-{down|up}
+       ;; aren't bound on VI. ideas?
+       "shift-pagedown" 'scroll-page-down-other-buffer
+       "shift-pageup" 'scroll-page-up-other-buffer
+       "shift-down" 'scroll-other-buffer-down
```
Comment (aartaka) [inline source/mode/prompt-buffer.lisp:43]: I wouldn't use `shift-down` -- it's too widespread.

Actually, we can ignore CUA bindings -- they are often too restrictive for what we're trying to do :P

File: source/mode/prompt-buffer.lisp [hunk 4]
```diff
@@ -36,6 +36,12 @@ Actions can be listed and run with `return-selection-over-action' (bound to
        "C-down" 'select-last
        "C-pagedown" 'select-next-source
        "C-pageup" 'select-previous-source
+       ;; scroll-page-{down|up}-other-buffer and scroll-other-buffer-{down|up}
+       ;; aren't bound on VI. ideas?
```
Comment (aartaka) [inline source/mode/prompt-buffer.lisp:40]: CC @Ambrevar

Comment (Ambrevar) [reply source/mode/prompt-buffer.lisp:40]: From https://stackoverflow.com/questions/9491013/scroll-the-other-window-in-vim-split, looks like there is no official Binding.

`M-j`/`M-k` or `C-M-j`/`C-M-k` could do.

**Comment by Ambrevar** [comment]
@aadcg A nit, but don't forget to name your PRs explicit and not (only) by references to something else.  For instance, you could name this one "[FOLLOW UP #2192] Improve hinting UI".

File: source/mode/element-hint.lisp (hunk 1 above)
Comment (aadcg) [reply source/mode/element-hint.lisp]: Sure! See 6b39e0e7.

File: source/mode/prompt-buffer.lisp (hunk 3 above)
Comment (aadcg) [reply source/mode/prompt-buffer.lisp]: Ignore all CUA bindings relative to scrolling the buffer behind the prompt correct? I'm fine with it!

See c7c92e13.

**Comment by aartaka** [comment]
Yes, looks fine to me.

**Comment by aadcg** [comment]
@Ambrevar ok to merge?

**Pull request merged**